flake.nix

```diff
@@ -1,40 +1,41 @@
 {
   inputs = {
-    naersk.url = "github:nix-community/naersk/master";
-    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    nixpkgs.url = "github:NixOS/nixpkgs?ref=nixpkgs-unstable";
+
+    naersk = {
+      url = "github:nix-community/naersk?ref=master";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
     utils.url = "github:numtide/flake-utils";
-    nixpkgs-mozilla = {
-      url = "github:mozilla/nixpkgs-mozilla";
-      flake = false;
+    rust-overlay = {
```

Comment:
Recommended replacement for nixpkgs-mozilla -> https://github.com/oxalica/rust-overlay/tree/master?tab=readme-ov-file#migration-from-nixpkgs-mozilla

Reply:
Yes I saw that was quite recent. I think #307 was a change in the right direction and I have maintained the fix by loading the `toolchain.toml` file.

@lxl66566 Would you be willing to try out this PR to see if it still works for you? I'm happy to work on it further, but I'm currently unable to build the flake from `master` so i would like to continue with this change :)

Reply:
Yes, it works on my machine

Reply:
Fantastic, thank you so much for checking!